### DISCUSSION TITLE
Zod schema hub like DefinitelyTyped?

### DISCUSSION THREAD

**Discussion by prescience-data** [colinhacks/zod#383] [Ideas]
I find myself creating lots of mirrors of TypeScript interfaces of popular libraries for parsing data into the correct shapes.

A thought that just occurred to me which seems like it would be super practically useful and potentially attract a lot of new users to Zod for the ergonomics is a hub like DefinitelyTyped `@types/<pkg>` monorepo. 

A suggestion would be to host it under a new `github.com/zod/schemas` org/repo, and publish under `@zod/<pkg>` or something generic like `@schemas/<pkg>`.

That way someone working with a popular lib could just pull in:

```json
{
  "dependencies": {
    "zod": "^3",
    "@schemas/ioredis": "^4.25.0",
    "@schemas/winston": "^3.3.3",
  }
}

``` 

and then use them immediately:

```typescript
import { Redis } from "ioredis"
import { RedisOptionsSchema } from "@schemas/ioredis"
import { myRedisOpts } from "../uncertain-module"

const redis = new Redis(RedisOptionsSchema.parse(myRedisOpts));

```

![magic](https://media1.tenor.com/images/9678f9af8d211fdcd77088948ed92bbf/tenor.gif)

**Comment by colinhacks** [comment]
This is certainly an interesting idea. I recently added an Ecosystem section to the README to support other Zod-related OSS projects. If you or others have implemented any Zod definition libraries, I'd recommend publishing it at `zod-ioredis` (as the case may be) for now and I'll link to it under "Ecosystem". If that list starts getting out of hand, we'll know there's enough interest to split this out into a dedicated namespace. I just reserved the NPM namespace "zodtypes" 👍

**Reply by prescience-data** [reply]
ah, `zodtypes` is definitely a cool namespace 🔥
